assets/js/googlesitekit/datastore/user/authentication.test.js

```diff
@@ -0,0 +1,220 @@
+				} ).not.toThrow();
+			} );
+		} );
+		describe( 'receiveConnection', () => {
+			it( 'requires the authentication param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).receiveAuthentication();
+				} ).toThrow( 'authentication is required.' );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getAuthentication', () => {
+			it( 'uses a resolver to make a network request', async () => {
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/user\/data\/authentication/
+					)
+					.mockResponseOnce(
+						JSON.stringify( responseAuthenticated ),
+						{ status: 200 }
+					);
+
+				const initialConnection = registry.select( STORE_NAME ).getAuthentication();
+				// The connection info will be its initial value while the connection
+				// info is fetched.
+				expect( initialConnection ).toEqual( undefined );
+				await subscribeUntil( registry,
+					() => (
+						registry.select( STORE_NAME ).getAuthentication() !== undefined
+					),
+				);
+
+				const authentication = registry.select( STORE_NAME ).getAuthentication();
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( authentication ).toEqual( responseAuthenticated );
+
+				const connectionSelect = registry.select( STORE_NAME ).getAuthentication();
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( connectionSelect ).toEqual( authentication );
+			} );
+
+			it( 'does not make a network request if data is already in state', async () => {
+				registry.dispatch( STORE_NAME ).receiveAuthentication( responseAuthenticated );
+
+				const authentication = registry.select( STORE_NAME ).getAuthentication();
+
+				await subscribeUntil( registry, () => registry
+					.select( STORE_NAME )
+					.hasFinishedResolution( 'getAuthentication' )
+				);
+
+				expect( fetch ).not.toHaveBeenCalled();
+				expect( authentication ).toEqual( responseAuthenticated );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
+				const response = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/user\/data\/authentication/
+					)
+					.mockResponseOnce(
+						JSON.stringify( response ),
+						{ status: 500 }
+					);
+
+				muteConsole( 'error' );
+				registry.select( STORE_NAME ).getAuthentication();
+				await subscribeUntil( registry,
+					// TODO: We may want a selector for this, but for now this is fine
+					// because it's internal-only.
+					() => store.getState().isFetchingAuthentication === false,
+				);
+
+				const connection = registry.select( STORE_NAME ).getAuthentication();
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( connection ).toEqual( undefined );
+			} );
+		} );
+		describe( 'isAuthenticated', () => {
+			it( 'uses a resolver get all authentication info', async () => {
```

Comment:
Description is a bit off-ish here.
```suggestion
			it( 'uses a resolver to load the authenticated value if not set yet', async () => {
```